aten/src/ATen/FunctionalInverses.cpp

```diff
@@ -0,0 +1,191 @@
+    int64_t start = 0;
+    for (auto i = 0; i < mutated_view_idx; ++i) {
+        start += split_sizes[i];
+    }
+    auto end = start + split_sizes[mutated_view_idx];
+    if (end > dim_size) end = dim_size;
+    return base.slice_scatter(mutated_view, dim, start, end, 1);
+}
+
+Tensor squeeze_inverse(const Tensor& base, const Tensor& mutated_view) {
+    return unsqueeze_to(mutated_view, base.sizes());
+}
+
+Tensor squeeze_inverse(const Tensor& base, const Tensor& mutated_view, int64_t dim) {
+    return unsqueeze_to(mutated_view, dim, base.sizes());
+}
+
+Tensor t_inverse(const Tensor& base, const Tensor& mutated_view) {
+    return mutated_view.t();
+}
+
+Tensor transpose_inverse(const Tensor& base, const Tensor& mutated_view, int64_t dim0, int64_t dim1) {
+    return mutated_view.transpose(dim0, dim1);
+}
+
+Tensor unsqueeze_inverse(const Tensor& base, const Tensor& mutated_view, int64_t dim) {
+    return mutated_view.squeeze(dim);
+}
+
+Tensor _indices_inverse(const Tensor& base, const Tensor& mutated_view) {
+    TORCH_INTERNAL_ASSERT(false, "Attempted to call _indices() during the functionalization pass. For now, sparse tensors aren't supported during functionalization");
+    return Tensor();
+}
+
+Tensor _values_inverse(const Tensor& base, const Tensor& mutated_view) {
+    TORCH_INTERNAL_ASSERT(false, "Attempted to call _values() during the functionalization pass. For now, sparse tensors aren't supported during functionalization");
+    return Tensor();
+}
+
+Tensor indices_inverse(const Tensor& base, const Tensor& mutated_view) {
+    TORCH_INTERNAL_ASSERT(false, "Attempted to call indices() during the functionalization pass. For now, sparse tensors aren't supported during functionalization");
+    return Tensor();
+}
+
+Tensor values_inverse(const Tensor& base, const Tensor& mutated_view) {
+    TORCH_INTERNAL_ASSERT(false, "Attempted to call values() during the functionalization pass. For now, sparse tensors aren't supported during functionalization");
+    return Tensor();
+}
+
+Tensor unbind_inverse(const Tensor& base, const Tensor& mutated_view, int64_t mutated_view_idx, int64_t dim) {
+    dim = at::maybe_wrap_dim(dim, base.sizes().size());
+    return base.select_scatter(mutated_view, dim, mutated_view_idx);
+}
+
+Tensor view_inverse(const Tensor& base, const Tensor& mutated_view, at::IntArrayRef size) {
+    return mutated_view.view(base.sizes());
+}
+
+Tensor view_inverse(const Tensor& base, const Tensor& mutated_view, at::ScalarType dtype) {
+    return mutated_view.view(base.scalar_type());
+}
+
+Tensor unfold_inverse(const Tensor& base, const Tensor& mutated_view, int64_t dimension, int64_t size, int64_t step) {
+    // I think autograd and the functionalization pass want the exact same thing here, but need to test to confirm.
+    return unfold_backward(mutated_view, base.sizes(), dimension, size, step);
+}
+
+Tensor alias_inverse(const Tensor& base, const Tensor& mutated_view) {
+    return mutated_view;
+}
```

Comment:
A simple, OpInfo style test for all view functions and their inverses showing that they uphold laws seems like a useful thing to do, and would help increase confidence here.

Reply:
I was kinda thinking of waiting to get this all hooked up to xla and running the existing view test suite to increase confidence. What do you think? I'm not sure of an easy way to add OpInfo tests that cover this, that doesn't involve either exposing all of the view inverse functions to python (requires more boilerplate), or creating some dummy tensor backend for testing that opts into functionalization and expose it to python.

Since I'm not sure when the "final" version of xla using this will happen (and the corresponding integration with ltc), I could probably pretty quickly throw up a PR with pytorch/xla that gets it to use the pass, and ensure that all of the view tests pass in the PR.

Reply:
> I'm not sure of an easy way to add OpInfo tests that cover this, that doesn't involve either exposing all of the view inverse functions to python (requires more boilerplate)

Oh, I guess I misunderstood, and thought you were planning to expose the view inverses in Python. I guess then we'll have to do E2E then...